compiler/rustc_middle/src/mir/interpret/allocation/provenance_map.rs

```diff
@@ -21,6 +22,20 @@ pub struct ProvenanceMap<Prov = AllocId> {
     bytes: Option<Box<SortedMap<Size, Prov>>>,
 }
 
+impl<D: Decoder, Prov: Decodable<D>> Decodable<D> for ProvenanceMap<Prov> {
+    fn decode(d: &mut D) -> Self {
+        Self { ptrs: Decodable::decode(d), bytes: None }
+    }
+}
+
+impl<S: Encoder, Prov: Encodable<S>> Encodable<S> for ProvenanceMap<Prov> {
+    fn encode(&self, s: &mut S) {
+        let Self { ptrs, bytes } = self;
+        debug_assert!(bytes.is_none());
```

Comment:
This is only true when `Prov` is `AllocId` (or more generally, when `Prov::OFFSET_IS_ADDR` is false). But we only need that instance. So maybe we can avoid making it generic?

Reply:
I tried that, but then we'd need to add manual impls for everything using `ProvenanceMap` (like `Allocation`, and I don't like that)